internal/uidriver/glfw/ui.go

```diff
@@ -487,27 +487,43 @@ func (u *UserInterface) adjustPosition(x, y int) (int, int) {
 	return x - int(ox/s), y - int(oy/s)
 }
 
-func (u *UserInterface) IsCursorVisible() bool {
+func (u *UserInterface) CursorMode() driver.CursorMode {
 	if !u.isRunning() {
-		return u.isInitCursorVisible()
+		return u.getInitCursorMode()
 	}
-	v := false
+	var v driver.CursorMode
 	_ = u.t.Call(func() error {
-		v = u.window.GetInputMode(glfw.CursorMode) == glfw.CursorNormal
+		switch u.window.GetInputMode(glfw.CursorMode) {
+		case glfw.CursorNormal:
+			v = driver.CursorModeVisible
+		case glfw.CursorHidden:
+			v = driver.CursorModeHidden
+		case glfw.CursorDisabled:
+			v = driver.CursorModeCaptured
+		default:
+			panic("Invalid cursor mode")
+		}
 		return nil
 	})
 	return v
 }
 
-func (u *UserInterface) SetCursorVisible(visible bool) {
+func (u *UserInterface) SetCursorMode(mode driver.CursorMode) {
 	if !u.isRunning() {
-		u.setInitCursorVisible(visible)
+		u.setInitCursorMode(mode)
 		return
 	}
 	_ = u.t.Call(func() error {
-		c := glfw.CursorNormal
-		if !visible {
+		c := glfw.NoAPI
+		switch mode {
+		case driver.CursorModeVisible:
+			c = glfw.CursorNormal
+		case driver.CursorModeHidden:
 			c = glfw.CursorHidden
+		case driver.CursorModeCaptured:
+			c = glfw.CursorDisabled
+		default:
+			panic("Invalid cursor mode")
```

Comment:
Please start with a small case (for consistency).

`fmt.Sprintf("invalid cursor mode: %d", mode)` is better.